accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,395 @@
+- Feature Name: Container as a Service Platform integration with Uyuni/SUSE Manager
+- Start Date: 2019-08-05
+- RFC PR: https://github.com/uyuni-project/uyuni-rfc/pull/10
+
+# Summary
+[summary]: #summary
+
+This RFC describes how Uyuni/SUSE Manager can be leveraged to perform cross-product integration with a cluster solution like Container as a Service Platform.
+
+# Motivation
+[motivation]: #motivation
+
+<!-- - Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions. -->
+
+Uyuni/SUSE Manager is the systems management tool and must offer to the end-user the possibility to deploy and manage other SUSE products like CaaS Platform.
+
+We will be focusing first on the following general scenarios:
+
+1. Deploy a cluster using Uyuni/SUSE Manager
+2. Manage a cluster using Uyuni/SUSE Manager
+
+The two subsequent CaaS Platform-specific sections will follow.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+
+## Generic design: integrate Uyuni/SUSE Manager with a cluster provider
+
+### Deploy a cluster using Uyuni/SUSE Manager
+
+Under Uyuni/SUSE Manager main menu, a new menu item "Clusters" should be introduced.
+All deployable and manageable cluster products (e.g. CaaSP, SOC, SES) will be in the sub-menu.
+
+Under that item, it should be possible to create, edit, and remove a cluster.
+
+To create a cluster, a user has to enter its name.
+
+Every detail about the cluster is saved into corresponding database tables (e.g. `susecluster`).
+
+API support for all the activities described below should be implemented as well as front-end support.
+
+#### Cluster requirements
+
+If the cluster required a specific set of nodes optionally with a defined configuration, the cluster provider must offer a way to deploy the whole infrastructure automatically, e.g. using AutoYaST profiles.
```

Comment:
> First disclaimer, I was told to review this in the context of SOC. I am not sure what this phrase means in that case. Could you clarify?

The intention is to have every cluster provider (in your case SOC) define a way to deploy the machines that will be comprising the cluster, e.g. by providing an AutoYaST profile.

> Next to the technology choice, I am concerned that deploying multiple products together doesn't "automatically" connect, and require user input (We cannot "automatically" assume things). What will happen in this case?

What do you mean? If a customer deploys two different products using SUSE Manager (e.g. SOC and CaaSP) then the products are not connected and should be connected manually, or we can think (as a far-future plan) to develop something in this area.

Reply:
> > The intention is to have every cluster provider (in your case SOC) define a way to deploy the machines that will be comprising the cluster, e.g. by providing an AutoYaST profile.
> 
> What about things on top? We just care about CaaSP being deployed, and we'll just deploy on top. I don't have the impression it's the scope of SUMA to know how ALL the applications in the world must be deployed. That sounds not scalable.

Exactly, SUMA does need (and does not want) to know what's running on top of the cluster. Do you think it's worth t explicitly point that out in the RFC?

> > What do you mean? If a customer deploys two different products using SUSE Manager (e.g. SOC and CaaSP) then the products are not connected and should be connected manually, or we can think (as a far-future plan) to develop something in this area.
> 
> Interconnecting openstack and ceph can be done in a multitude of ways depending on the customer expectations. Do you really want to develop something in that area? This means having deep knowledge of both SES and SOC. That sounds by far outside the scope of SUMA. But if you have the time and bandwith to do so, happy to see that happening. I just don't think it's how it should be done. SUMA should deploy the base system/cluster (for example handling bare metal/configuring host for CaaSP/eventually deploy caasp, and leave the rest to the other teams, because it then starts to be at the "app" level).

I think the far-future plan should not be discussed in the current RFC, it's just an idea of what we can might expect from the future developments. Do you agree?